health-services/project-factory/src/server/validators/microplanValidators.ts

```
@@ -0,0 +1,300 @@
+    const boundaryColumn = getLocalizedName("HCM_ADMIN_CONSOLE_RESIDING_BOUNDARY_CODE_MICROPLAN", localizationMap);
+    if (!item?.[boundaryColumn]) {
+        errors.push({ status: "INVALID", rowNumber: item?.["!row#number!"], errorDetails: `Data in ${boundaryColumn} column can’t be empty, please update the data and re-upload` })
+    }
+}
+
+export function validateFacilityBoundaryForLowestLevel(request: any, boundaries: any, rowData: any, errors: any = [], localizationMap?: { [key: string]: string }) {
+    if (request?.body?.ResourceDetails?.type == "facility" && request?.body?.ResourceDetails?.additionalDetails?.source == "microplan") {
+        const hierarchy = request?.body?.hierarchyType?.boundaryHierarchy
+        const lastLevel = hierarchy?.[hierarchy.length - 1]?.boundaryType
+        for (const data of rowData?.boundaryCodes) {
+            const boundaryFromBoundariesType = boundaries.find((boundary: any) => boundary.code == data)?.type
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)